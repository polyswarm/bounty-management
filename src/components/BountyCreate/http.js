class Http {
  constructor(url) {
    this.url = url;
    this.xhr = null;
  }

  cancel() {
    const { xhr } = this;
    if (xhr) {
      xhr.abort();
    }
  }

  uploadFiles(files) {
    return new Promise((resolve, reject) => {
      const url = this.url+'/artifacts';
      if(this.url && files && files.length > 0) {
        // Add files
        const formData = new FormData();
        files.forEach((file) => {
          formData.append(file.name, file, file.name);
        });

        // open connection
        const xhr = new XMLHttpRequest();

        // attach listeners
        xhr.onerror = () => reject(xhr.statusText);
        xhr.onload = () => resolve(xhr.response);

        // send to server
        xhr.open('post', url);
        xhr.send(formData);
        this.xhr = xhr;
      } else {
        reject('No URL set');
      }
    })
      .then(body => body.json)
      .then(json => json.artifacts);
  }

  uploadBounty(amount, artifacts, duration) {
    const url = this.url + '/bounties';
    return new Promise((resolve, reject) => {
      if(this.url && amount && duration && artifacts && artifacts.length > 0) {
        const bounty = {
          amount: amount,
          duration: duration,
          artifacts: artifacts
        };
        resolve(bounty);
      } else {
        reject('Invalid bounty.');
      }
    })
      .then(bounty => fetch(url, {
        method: 'post',
        body: bounty,
      }))
      .then(response => {
        if (response.ok) {
          return response;
        }
        throw new Error('Error response was not ok');
      })
      .then(response => response.json())
      .then(body => body.guid);
  }
}
export default Http;
