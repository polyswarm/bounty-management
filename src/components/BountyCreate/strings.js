const strings = {
  createBounty: 'Create Bounty',
};
export default strings;
