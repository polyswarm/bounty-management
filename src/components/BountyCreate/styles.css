.Bounty-Create {
  background-color: #fafafa;
  border-radius: 20px;
  display: block;
  margin: 0 auto;
  max-width: 768px;
  position: relative;
}

.Container {
  display: grid;
  grid-column-gap: 28px;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: 400px 35px 40px;
  padding: 20px;
}

.Bounty-Create-Error {
  color: red;
  grid-column: 1 / 3 ;
  grid-row: 2;
  width: 100%;
}

.Button {
  grid-column: 1/ 3;
  grid-row: 3;
  margin-top: 0;
  width: 100%;
}
