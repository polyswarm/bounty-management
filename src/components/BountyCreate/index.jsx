// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import DropTarget from '../DropTarget';
import FileList from '../FileList';
import Button from '../Button';
import ModalPassword from '../ModalPassword';
// Component imports
import strings from './strings';
import './styles.css';
import Http from './http';

class BountyCreate extends Component {
  constructor(props) {
    super(props);
    this.state = {
      files: [],
      uploading: false,
      error: null,
    };
    this.onMultipleFilesSelected = this.onMultipleFilesSelected.bind(this);
    this.onFileRemoved = this.onFileRemoved.bind(this);
    this.createBounty = this.createBounty.bind(this);
    this.onClickHandler = this.onClickHandler.bind(this);
    this.onClearAll = this.onClearAll.bind(this);
    this.cancel = this.cancel.bind(this);
    this.onWalletChangeHandler = this.onWalletChangeHandler.bind(this);
  }

  componentDidMount() {
    const { props: { url } } = this;
    this.http = new Http(url);
  }

  render() {
    const { state: { files, uploading, error } } = this;
    const { props: { url, walletList } } = this;
    return(
      <div className='Bounty-Create'>
        <ModalPassword ref={(modal) => this.modal = modal}
          url={url}
          walletList={walletList}
          onWalletChange={this.onWalletChangeHandler} />
        <div className='Container'>
          <FileList files={files}
            clear={this.onClearAll}
            removeFile={this.onFileRemoved}/>
          <DropTarget onFilesSelected={this.onMultipleFilesSelected}/>
          {error && (
            <p className='Bounty-Create-Error'>{error}</p>
          )}
          <Button
            className='Bounty-Create-Upload'
            disabled={uploading || !files || files.length === 0 }
            onClick={this.onClickHandler}>
            {uploading && (strings.cancel)}
            {!uploading && (strings.createBounty)}
          </Button>
        </div>
      </div>
    );
  }

  onMultipleFilesSelected(files) {
    const f = this.state.files.slice();
    const combined = f.concat(files);
    this.setState({ files: combined });
  }

  onFileRemoved(index) {
    const files = this.state.files.slice();
    if (index >= 0 && index < files.length) {
      files.splice(index, 1);
      this.setState({ files: files, error: null, });
    }
  }

  onClickHandler() {
    const { state: { uploading } } = this;
    if (uploading) {
      this.cancel();
    } else {
      this.modal.open();
      // if (!isUnlocked) {
      //   this.modal.open();
      // } else {
      //   this.createBounty();
      // }
    }
  }

  onWalletChangeHandler(store) {
    const { props: { onWalletChange } } = this;
    if (onWalletChange) {
      onWalletChange(store);
    }
    this.createBounty();
  }

  onClearAll() {
    this.setState({ files: [], error: null, });
  }

  cancel() {
    const { http } = this;
    http.cancel();
  }

  createBounty() {
    const { props: { addBounty } } = this;
    const { state: { uploading} } = this;
    const files = this.state.files.slice();

    const http = this.http;
    if (!uploading && files && files.length > 0) {
      this.setState({uploading: true, error: null});
      http.uploadFiles(files)
        .then((artifact) => http.uploadBounty('6250000000000000000', artifact, 300))
        .then(result => {
          if (addBounty) {
            addBounty(result);
          }
        })
        .catch((error) => {
          let errorMessage;
          const { props: { onWalletChange } } = this;
          onWalletChange(false);
          if (!error || error.length === 0) {
            errorMessage = strings.error;
          } else {
            errorMessage = error.message;
          }
          this.setState({error: errorMessage});
        })
        .then(() => this.setState({
          uploading: false,
        }));
    }
  }
}

BountyCreate.propTypes = {
  isUnlocked: PropTypes.bool,
  walletList: PropTypes.array,
  onWalletChange: PropTypes.func,
  addBounty: PropTypes.func,
  url: PropTypes.string,
};
export default BountyCreate;
