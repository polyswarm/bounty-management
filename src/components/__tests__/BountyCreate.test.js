import React from 'react';
import {shallow, render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import BountyCreate from '../BountyCreate';
import Http from '../BountyCreate/http';

const mockUploadFiles = jest.fn().mockImplementation(() => {
  return new Promise(resolve => {
    resolve();
  });
});

const mockUploadBounty = jest.fn().mockImplementation(() => {
  return new Promise(resolve => {
    resolve();
  });
});

jest.mock('../BountyCreate/http', () => {
  // Works and lets you check for constructor calls:
  return jest.fn().mockImplementation(() => {
    return {
      uploadFiles: mockUploadFiles,
      uploadBounty: mockUploadBounty,
    };
  });
});

beforeEach(() => {
  jest.clearAllMocks();
  Http.mockClear();
  mockUploadFiles.mockClear();
  mockUploadBounty.mockClear();
});

it('renders without crashing', () => {
  const wrapper = render(<BountyCreate />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('deletes the index 0 when onFileRemoved called', () => {
  const wrapper = shallow(<BountyCreate />);
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  instance.setState({files});

  //act
  instance.onFileRemoved(0);
  expect(instance.state.files).toHaveLength(1);
  expect(instance.state.files).toEqual([{name: 'omed'}]);
});

it('deletes the file at index when onFileRemoved called', () => {
  const wrapper = shallow(<BountyCreate />);
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  instance.setState({files});

  //act
  instance.onFileRemoved(1);
  expect(instance.state.files).toHaveLength(1);
  expect(instance.state.files).toEqual([{name: 'demo'}]);
});

it('deletes all files when onClearAll is called', () => {
  const wrapper = shallow(<BountyCreate />);
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  instance.setState({files});

  //act
  instance.onClearAll();

  expect(instance.state.files).toHaveLength(0);
  expect(instance.state.files).toEqual([]);
});

it('doesn\'t delete the file at index when out of bounds', () => {
  const wrapper = shallow(<BountyCreate />);
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  instance.setState({files});

  //act
  instance.onFileRemoved(1000);
  expect(instance.state.files).toHaveLength(2);
  expect(instance.state.files).toEqual([{name: 'demo'}, {name: 'omed'}]);
});

it('doesn\'t delete the file at index when negative', () => {
  const wrapper = shallow(<BountyCreate />);
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  instance.setState({files});

  //act
  instance.onFileRemoved(-1);
  expect(instance.state.files).toHaveLength(2);
  expect(instance.state.files).toEqual([{name: 'demo'}, {name: 'omed'}]);
});

it('doesn\'t throw with an empty file array when onFileRemoved called', () => {
  const wrapper = shallow(<BountyCreate />);
  const instance = wrapper.instance();

  //act
  instance.onFileRemoved(0);
});

it('stores additional files in state.files', () => {
  const wrapper = shallow(<BountyCreate />);
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files});

  //act
  instance.onMultipleFilesSelected([{name: 'asdf'}]);
  expect(instance.state.files).toHaveLength(3);
  expect(instance.state.files).toEqual([{name: 'demo'}, {name: 'omed'}, {name: 'asdf'}]);
});

it('calls uploadFiles when all parameters are met (files, addBounty, url)', () => {
  const addBounty = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      addBounty={addBounty}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty();

  // assert
  expect(mockUploadFiles).toHaveBeenCalledTimes(1);
});

it('doesn\'t call uploadFiles when parameters are missing', () => {
  const wrapper = mount(<BountyCreate />);
  const instance = wrapper.instance();

  // act
  instance.createBounty();
  // assert
  expect(mockUploadFiles).toHaveBeenCalledTimes(0);
});

it('doesn\'t call uploadBounty when uploadFiles fails', () => {
  const mockBadUploadFiles = jest.fn().mockImplementation(() => {
    return new Promise((resolve, reject) => {
      reject();
    });
  });
  Http.mockImplementation(() => {
    return {
      uploadFiles: mockBadUploadFiles,
      uploadBounty: mockUploadBounty,
    };
  });

  const addBounty = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      addBounty={addBounty}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty();

  // assert
  expect(mockUploadBounty).toHaveBeenCalledTimes(0);
});

it('calls uploadBounty when uploadFiles succeeds', (done) => {
  const mockGoodUploadFiles = jest.fn().mockImplementation(() => {
    return new Promise((resolve) => {
      resolve(['demo', 'asdf']);
    });
  });
  Http.mockImplementation(() => {
    return {
      uploadFiles: mockGoodUploadFiles,
      uploadBounty: mockUploadBounty,
    };
  });

  const addBounty = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      addBounty={addBounty}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty()
    .then(() => {

    // assert
      try {
        expect(mockUploadBounty).toHaveBeenCalledWith('62500000000000000', ['demo', 'asdf'], 300);
        expect(mockUploadBounty).toHaveBeenCalledTimes(1);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
});

it('calls addBounty when upload bounty is a success', (done) => {
  const mockGoodUploadFiles = jest.fn().mockImplementation(() => {
    return new Promise((resolve) => {
      resolve(['demo', 'asdf']);
    });
  });
  const mockGoodUploadBounty = jest.fn().mockImplementation(() => {
    return new Promise((resolve) => {
      resolve('asdf');
    });
  });
  Http.mockImplementation(() => {
    return {
      uploadFiles: mockGoodUploadFiles,
      uploadBounty: mockGoodUploadBounty,
    };
  });

  const addBounty = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      addBounty={addBounty}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty()
    .then(() => {
    // assert
      try {
        expect(addBounty).toHaveBeenCalledWith('asdf');
        expect(addBounty).toHaveBeenCalledTimes(1);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
});

it('sets errors to null when uploads complete', (done) => {
  const addBounty = jest.fn();
  const onWalletChange = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      onWalletChange={onWalletChange}
      walletList={walletList}
      addBounty={addBounty}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files});
  const setStateMock = jest.spyOn(BountyCreate.prototype, 'setState');

  instance.createBounty()
    .then(() => {

      // assert
      try {
        expect(setStateMock).toHaveBeenLastCalledWith({error: null, files: []});
        expect(setStateMock).toHaveBeenCalledTimes(1);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
});

it('has uploading true after calling createBounty', (done) => {
  const setState = jest.spyOn(BountyCreate.prototype, 'setState');
  const addBounty = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      addBounty={addBounty}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files});
  setState.mockClear();

  // act
  instance.createBounty().then(() => {

  // assert
    try {
      expect(setState.mock.calls[0][0]).toEqual({error: null, files:[]});
      done();
    } catch (error) {
      done.fail(error);
    }
  });
});

it('disables button when there are no files', () => {
  const wrapper = mount(<BountyCreate url={'url'}/>);
  const files = [
  ];
  wrapper.setState({files: files, uploading: false});

  expect(wrapper.find('.Bounty-Create-Upload').props().disabled).toBeTruthy();
});

it('enables button when there are files, a url and not uploading ', () => {
  const wrapper = mount(<BountyCreate url={'url'}/>);
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files, uploading: false});
  expect(wrapper.find('.Bounty-Create-Upload').props().disabled).toBeFalsy();
});

it('shows the error when state has an error', () => {
  const wrapper = mount(<BountyCreate url={'url'}/>);
  wrapper.setState({error: 'Error'});

  expect(wrapper.find('.Bounty-Create-Error').props().children).toBe('Error');
});

it('opens the modal if isUnlocked not set on create click', () => {
  const onWalletChange = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      onWalletChange={onWalletChange}
      walletList={walletList}
      isUnlocked={false}/>
  );
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files, uploading: false});

  wrapper.find('.Bounty-Create-Upload').simulate('click');

  expect(wrapper.find('.ModalContent')).toHaveLength(1);
});

it('calls create after modal is successfully closed & returns unlocked', () => {
  const createBounty = jest.spyOn(BountyCreate.prototype, 'createBounty');
  const onWalletChange = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      onWalletChange={onWalletChange}
      walletList={walletList}
      isUnlocked={false}/>
  );
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files, uploading: false});
  const instance = wrapper.instance();

  instance.onWalletChangeHandler(true, false);

  expect(createBounty).toHaveBeenCalledTimes(1);
});

it('calls onWalletChange when modal closed and password checked', () => {
  const onWalletChange = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      onWalletChange={onWalletChange}
      walletList={walletList}
      isUnlocked={false}/>
  );
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files, uploading: false});
  const instance = wrapper.instance();

  instance.onWalletChangeHandler(true, true);

  expect(onWalletChange).toHaveBeenCalledTimes(1);
  expect(onWalletChange).toHaveBeenCalledWith(true);
});

it('calls onWalletChange when modal closed and password not checked', () => {
  const onWalletChange = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      onWalletChange={onWalletChange}
      walletList={walletList}
      isUnlocked={false}/>
  );
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files, uploading: false});
  const instance = wrapper.instance();

  instance.onWalletChangeHandler(false, false);

  expect(onWalletChange).toHaveBeenCalledTimes(1);
  expect(onWalletChange).toHaveBeenCalledWith(false);
});

it('calls onWalletChange with false when upload bounty returns 401', (done) => {
  const mockBadUploadBounty = jest.fn().mockImplementation(() => {
    return new Promise((resolve, reject) => {
      const error = {
        status: 401,
      };
      reject(error);
    });
  });
  Http.mockImplementation(() => {
    return {
      uploadFiles: mockUploadFiles,
      uploadBounty: mockBadUploadBounty,
    };
  });

  const addBounty = jest.fn();
  const onWalletChange = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      onWalletChange={onWalletChange}
      addBounty={addBounty}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty()
    .then(() => {
      // assert
      try {
        expect(onWalletChange).toHaveBeenCalledTimes(1);
        expect(onWalletChange).toHaveBeenCalledWith(false);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
});

it('calls on error when something goes wrong in the upload', (done) => {
  const mockBadUploadBounty = jest.fn().mockImplementation(() => {
    return new Promise(() => {
      throw Error('Failed.');
    });
  });
  Http.mockImplementation(() => {
    return {
      uploadFiles: mockUploadFiles,
      uploadBounty: mockBadUploadBounty,
    };
  });

  const addBounty = jest.fn();
  const onWalletChange = jest.fn();
  const onError = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      onWalletChange={onWalletChange}
      addBounty={addBounty}
      onError={onError}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty()
    .then(() => {

    // assert
      try {
        expect(onError).toHaveBeenCalledTimes(1);
        expect(onError).toHaveBeenCalledWith('Failed.');
        done();
      } catch (error) {
        done.fail(error);
      }
    });
});

it('should call addCreateBountyRequest and removeCreateBountyRequest in createBounty', (done) => {
  const addCreateBountyRequest = jest.spyOn(BountyCreate.prototype, 'addCreateBountyRequest');
  const removeCreateBountyRequest = jest.spyOn(BountyCreate.prototype, 'removeCreateBountyRequest');
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty()
    .then(() => {

      // assert
      try {
        expect(addCreateBountyRequest).toHaveBeenCalledTimes(1);
        expect(removeCreateBountyRequest).toHaveBeenCalledTimes(1);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
});

it('should call addRequest and removeRequest in createBounty', (done) => {
  const addRequest = jest.fn();
  const removeRequest = jest.fn();
  const walletList = [];
  const wrapper = mount(
    <BountyCreate url={'url'}
      walletList={walletList}
      addRequest={addRequest}
      removeRequest={removeRequest}
      isUnlocked={true}/>
  );
  const instance = wrapper.instance();
  const files = [
    {name: 'demo'},
    {name: 'omed'},
  ];
  wrapper.setState({files: files});

  // act
  instance.createBounty()
    .then(() => {

      // assert
      try {
        expect(addRequest).toHaveBeenCalledTimes(1);
        expect(removeRequest).toHaveBeenCalledTimes(1);
        done();
      } catch (error) {
        done.fail(error);
      }
    });
});