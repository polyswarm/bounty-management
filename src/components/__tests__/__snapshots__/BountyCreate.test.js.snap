// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Bounty-Create"
>
  <div
    class="ModalPassword"
  >
    <div>
      
    </div>
  </div>
  <div
    class="Container"
  >
    <div
      class="File-List"
    >
      <header
        class="File-List-Header"
      >
        <p
          class="File-List-Title"
        >
          0 Files
        </p>
        <button
          class="Button"
          disabled=""
        >
          - Clear
        </button>
      </header>
      <ul
        class="List"
      >
        <div>
          
        </div>
      </ul>
    </div>
    <div
      class="Drop-Target"
    >
      <div>
        <p>
          Drag and drop your files here, or 
        </p>
        <form>
          <input
            class="hidden"
            id="file"
            multiple=""
            type="file"
          >
            
          </input>
          <label
            for="file"
          >
            click here.
          </label>
        </form>
      </div>
    </div>
    <button
      class="Button"
      disabled=""
    >
      Create Bounty
    </button>
  </div>
</div>
`;
